client/signup/steps/domains/index.jsx

```diff
@@ -310,7 +304,8 @@ class DomainsStep extends Component {
 			Object.assign(
 				{ domainItem },
 				this.isDependencyShouldHideFreePlanProvided() ? { shouldHideFreePlan } : {},
-				useThemeHeadstartItem
+				useThemeHeadstartItem,
+				signupDomainOrigin ? { signupDomainOrigin } : {}
```

Comment:
Nit: can probably be simplified with.

```suggestion
				...( signupDomainOrigin && { signupDomainOrigin } ) 
```